### PR TITLE
[nrf fromtree/fromlist] add dmm component

### PR DESCRIPTION
**Pull request by nika-nordic**
Upstream PR: https://github.com/zephyrproject-rtos/zephyr/pull/71278
manifest-pr-skip

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi [hunk 1]
```diff
@@ -26,8 +26,16 @@
 				reg = <0x800 DT_SIZE_K(2)>;
 			};
 
-			cpuapp_data: memory@1000 {
-				reg = <0x1000 DT_SIZE_K(256)>;
+			cpuapp_cpusec_misc_shm: memory@1000 {
```
Comment (nika-nordic) [inline boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi:29]: @hubertmis should we disable this region by default?

Comment (hubertmis) [reply boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi:29]: I understand this region is needed for IPC with SecDom. Let's keep it enabled by default, similarly to `cpusec_cpuapp_ipc_shm`.

**Review by hubertmis — APPROVED** [review]

**Review by osaether — APPROVED** [review]
LGTM

**Comment by nika-nordic** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/74302#issuecomment-2500009338

**Pull request closed without merging**